pkg/indexers/indexers.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package indexers
+
+import (
+	"github.com/kcp-dev/logicalcluster"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clusters"
+)
+
+const (
+	// ByLogicalCluster is the name for the index that indexes by an object's logical cluster.
+	ByLogicalCluster = "kcp-global-byLogicalCluster"
```

Comment:
we call this ByWorkspace everywhere else.

What is kcp-global? A new convention? Why not call it just `byWorkspace` and `byWorkspaceAndNamespace` ? I think we really want an indexers package with helpers to add those indexers without falling over each other when they are added multiple times.

Reply:
oh, this is it :)

Reply:
Then, can we add the mentioned helper here instead of just referencing `ClusterScoped` and `NamespaceScoped`?

Reply:
Yes. And once the new informers&listers are ready, we'll do a full cleanup.